### PR TITLE
Master

### PR DESCRIPTION
**Pull request by k3b**
* #379 moved from osmdroid-android/src/main/java/org/osmdroid/api/ to osmdroid-googlemaps-test/src/main/java/org/osmdroid/api/
* Added (Clickable)IconOverlay. 
* Updated inline javadoc-s (text+uml-classdiagram).
* #394: Added AsyncTaskDemoFragment
* Update to mapsforge-0.6.1; 
* Fix mapsforge/mapsforge#868: added more temporary memory for rendering.
* Added debuglogging to osmdroid-mapsforge via OpenStreetMapTileProviderConstants.DEBUG_TILE_PROVIDERS==true

### CODE REVIEW

File: osmdroid-android/build.gradle [hunk 1]
```diff
@@ -2,6 +2,10 @@ description = 'OSMdroid Android'
 apply plugin: 'com.android.library'
 apply from: 'https://raw.githubusercontent.com/chrisdoyle/gradle-fury/master/gradle/android-support.gradle'
 
+configurations {
```
Comment (spyhunter99) [inline osmdroid-android/build.gradle]: not sure why this is here, you can generate javadocs with
gradlew install -Pprofile=javadoc

File: osmdroid-android/build.gradle [hunk 2]
```diff
@@ -18,4 +22,16 @@ android {
...
 dependencies {
     testCompile group: 'junit', name: 'junit', version: '4.8.2'
+//    javadocDeps 'com.android.support:support-annotations:22.2.0'
+}
+
+task _javadoc(type: Javadoc) {
+    source = android.sourceSets.main.java.srcDirs
+    classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
+    classpath += configurations.javadocDeps
+    destinationDir = file("../javadoc/")
```
Comment (spyhunter99) [inline osmdroid-android/build.gradle]: oh is this to include the images in the javadoc jar? awesome!

**Comment by spyhunter99** [comment]
+10000 points for the javadoc documentation
-2 points, next time do one issue at a time per pull request. slamming multiple issues together on a single branch makes it hard to differentiate commits.

**Comment by spyhunter99** [comment]
whats the different between ItemizedIconOverlay and ClickableIconOverlay? Also the Marker class is an overlay that provides onclick methods (not a template class though)

**Comment by spyhunter99** [comment]
AsyncTaskDemoFragment - i like it!

**Comment by k3b** [comment]
my local-k3b-devbrach build.gradle files are different from the official master-build.gradle (im have removed gradle-fury and appcompat): 

* I cannot build offline (without internet connection) when working in a train so i added my own _doc task. you can delete it from master branch
* (Clickable)IconOverlay (created in 2015) are copied from my photo project: 
  * you cannot create Marker-s in async task because of resouce-access (#397)  
  * ClickableIconOverlay can work with any kind of data. In my photoapp i use it similar to cursor-list-adapter where text, image, ... come from database . In ItemizedIconOverlay  the data is stored in the items

@spyhunter99 - AsyncTaskDemoFragment: Do you have any idea why after `zoom + recreate items + mapView.invalidate()` not all items are visible? my photoapp has the same issue

**Comment by spyhunter99** [comment]
regarding the zoom issues, it looks like some of your icon calculations are off by a bit.  Apparently its' only on certain zoom levels (at least for me in the emulator). Maybe try adding a 40 pixel buffer to the edges of the screen before calculating the points

**Comment by spyhunter99** [comment]
pushed to branch feature/PR398 after fixing the merge issues

**Comment by spyhunter99** [comment]
you can follow the build here: https://travis-ci.org/osmdroid/osmdroid/builds/158145076

**Comment by k3b** [comment]
may be i found the zoom issue: 

		private void reloadMarker() {
			// initialized
                        // if there is no background load active
			if (currentBackgroundMarkerLoaderTask == null) {
				// not active yet
				// List<Overlay> oldItems = mFolderOverlaySummaryMarker.getItems();

				int zoom = this.mMapView.getZoomLevel();
				BoundingBox world = this.mMapView.getBoundingBox();

				reloadMarker(world, zoom);
			}
                        // else: lost zoom/scroll visible part has changed: todo start reload again
		}

I will analyse this this evening

**Comment by spyhunter99** [comment]
@k3b so should we close this one and open a new PR for the other branch?

**Pull request closed without merging**